Sources/LeafKit/LeafSerializer.swift

```diff
@@ -137,8 +137,9 @@ struct LeafSerializer {
         for (idx, item) in array.enumerated() {
             var innerContext = self.data
             
-            if idx == 0 { innerContext["isFirst"] = .bool(true) }
-            else if idx == array.count - 1 { innerContext["isLast"] = .bool(true) }
+            innerContext["isFirst"] = .bool(idx == 0)
```

Comment:
Use `idx == array.startIndex` for `Collection` correctness.

Reply:
Updated